app/models/provided/exchange_rate.rb

```diff
@@ -0,0 +1,22 @@
+class Provided::ExchangeRate
+  include Providable, Retryable
+
+  def initialize
+    @provider = exchange_rates_provider
+  end
+
+  def fetch(from:, to:, date:)
+    retrying Provider::Base.known_transient_errors do
+      response = provider.fetch_exchange_rate(from:, to:, date:)
+
+      ExchangeRate.new \
+        base_currency: from,
+        converted_currency: to,
+        rate: response.rate,
+        date: date
+    end
+  end
```

Comment:
You know? Come to think of it, I think this abstraction might be anemic.

I think we can just call the provider and instantiate a record from within the model. Would probably live within its own concern.

This would happen inside `ExchangeRate.fetch_from_provider` and `ExchangeRate.fetch_historical_from_provider` methods.

I'll spike on it tonight and see where it gets us. Will probably move the retryable stuff to the provider instead.

Reply:
https://github.com/maybe-finance/maybe/pull/561/commits/c49782ebfab9289bfd871c4616bd64e48fa063b8